src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,31 @@
+import Popup from "./Popup";
+
+export default class PopupWithConfirmation extends Popup {
+  constructor(popupSelector) {
+    super({ popupSelector });
+    this._popupForm = this._popupElement.querySelector(".modal__form");
+    this._submitButton = this._popupForm.querySelector(".modal__button");
+    this._submitButtonText =
+      this._submitButton.querySelector("#delete-card-modal");
+  }
+
+  setSubmitAction(action) {
+    this._handleFormSubmit = action;
+  }
+
+  setLoading(submit, loadingText = "Saving...") {
+    if (submit) {
+      this._submitButton.textContent = loadingText;
+    } else {
+      this._submitButton.textContent = this._submitButtonText;
```

Comment:
* The confirmation button doesn't have any text after the first submition http://joxi.ru/EA4YWBvS03OG9r